app/controllers/api/v2/table_preferences_controller.rb

```diff
@@ -36,11 +40,12 @@ def create
       api :PUT, "/users/:user_id/table_preferences/:name", N_("Updates a table preference for a given table")
       param_group :table_preference
       def update
-        process_response @table_preference.update(:columns => params[:columns])
+        process_response @table_preference.update(:name => params[:name], :columns => params[:columns])
```

Comment:
I don't think it makes sense to allow users change the name of the table, since it would lead to a need to change columns as well (most of the time). The only use case for it I could think of is to change the name in case it was wrongly spelled (e.g. `host` instead of `hosts`). And even in this case, we should probably show the list of supported tables explicitly, so users don't need to guess what they need to send.